### PR TITLE
MNTOR-1038 refactor email utility for V2

### PR DESCRIPTION
**Pull request by rhelmer**
# References: 
Jira: MNTOR-1038

# Checklist (Definition of Done)
- [x] Localization strings (if needed) have been added.
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] I've added a unit test to test for potential regressions of this bug.
- [x] Product Owner accepted the User Story (demo of functionality completed) or waived the privilege.
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [ ] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

File: src/utils/email.js
Comment (rhelmer) [inline src/utils/email.js:102]: Do we want to try to tackle this now? I'd like more details on the problem.

File: src/utils/email.js
Comment (rhelmer) [inline src/utils/email.js:13]: Is this still preferred over `console.*`? I noticed there's a `log.js` in `./utils` so I assumed so, but I see we're using the former in some places in `./src`.

File: src/utils/email.js
Comment (rhelmer) [inline src/utils/email.js:47]: https://github.com/mozilla/blurts-server/pull/2765

File: src/utils/email.js
Comment (rhelmer) [reply src/utils/email.js:19]: Is this something we can have tooling do? eslint or prettier etc?

Comment (rhelmer) [reply src/utils/email.js:19]: Oh OK I misread at first that it was just styling :) Yes I think this could use named exports of functions vs. exporting the whole object.

File: src/utils/sha1.js
Comment (rhelmer) [reply src/utils/sha1.js]: Oh OK I didn't see it was already exported from there - that's fine for now, it does seem a little unrelated to FxA proper but not a big deal, I'd rather file an issue for later if we want to do this.

File: src/utils/email.js
Comment (rhelmer) [reply src/utils/email.js:23]: @mansaj oh, hm so the reason I did this is that the current design of `EmailUtils` is a singleton, so `init` is called once and then the other functions use `this.*` as appropriate.

I'm not actually sure this is necessary (or desirable) though, I'm going to make this change but leave this comment unresolved for the moment while I look into it.

**Review by mansaj — APPROVED** [review]
LGTM, pending unsub discussion

File: package.json
Comment (toufali) [inline package.json:53]: Do we want to include this as part of `src/package.json` or is intended to live in the root (legacy) package?  (We _will_ have to reconcile both files, but it might be nice for new stuff to live in the new package?)

File: src/package.json
Comment (toufali) [inline src/package.json:46]: I see `quibble` added but not used – is it safe to remove?

**Review by toufali — COMMENTED** [review]
Thanks for this refactor @rhelmer!  Tests work as expected locally.  
- There's a readme section (Trigger Breach Alert Email) that involves this utility – does it need updating as part of this PR?
- It might be nice to align on code style – it could simplify reviews and make git blame work better!

**Comment by toufali** [comment]
RE Unsubscribe – it's currently implemented for the monthly emails only, and done in a "quick and dirty" way.  I had a discussion with PM that we should have a more established "subscriptions" section in our settings page, but that has been de-prioritized for now, so we can stick with what we've got!

File: src/package.json
Comment (rhelmer) [reply src/package.json:46]: Hm, that's a dependency of `testdouble` (`quibble` is a module loader, it's a lot like `esmock`, but these aren't very useful without a mocking library like `testdouble` or similar).

I don't think it needs to be in the `package.json` though, this is probably a remnant of when I was exploring using `esmock` and then tried `quibble` directly, and then realized `testdouble` was closer to what we needed to replace `jest`. I'll see if I can remove it from here and just have it as a dep of td in `package-lock.json`

**Comment by rhelmer** [comment]
> Thanks for this refactor @rhelmer! Tests work as expected locally.
> 
>     * There's a readme section (Trigger Breach Alert Email) that involves this utility – does it need updating as part of this PR?

Thanks for pointing this out - I think it will need updating, but two things make me thing we're not ready yet:

1. I'm not sure we've implemented enough in V2 to be able to complete the test in that section (have we ported the breach alerts template yet?)
2. I feel like we should start explicitly defaulting to V2, otherwise it's hard to tell which version the readme refers to

I think this is important though, could we file follow-up tickets? I feel like we should have one to convert the breach alerts template (if we haven't done that already) and we should explicitly flip V2 on by default (blocked by the current V2 staging work).

I didn't want to tackle converting all the templates over in this PR because I have still need to understand better how to structure these, and I want to unblock the settings back-end and front-end PRs in the meantime.

Also as an aside, on re-reading this section I just noticed the mention in the readme that Mailinator makes it easier to test images unlike the way I've been testing, that makes thing much easier for me!

>     * It might be nice to align on code auto-formatting – it could simplify reviews and make git blame work better!

Yes you mentioned this before and I meant to follow up before now, sorry about that! I just took a few minutes to look at my setup, and I think I was assuming we were using a shared config already. Is it possible to take the settings in the readme and put them into `./vscode/settings.json`?

BTW I'm happy to file the issues if that sounds correct and you agree that these are OK to break out as followup tasks ^

**Comment by rhelmer** [comment]
> > ```
> > * It might be nice to align on code auto-formatting – it could simplify reviews and make git blame work better!
> > ```
> 
> Yes you mentioned this before and I meant to follow up before now, sorry about that! I just took a few minutes to look at my setup, and I think I was assuming we were using a shared config already. Is it possible to take the settings in the readme and put them into `./vscode/settings.json`?

OK I went ahead and fixed up my local environment, and I opened this WIP PR as an example of how we could share: https://github.com/mozilla/blurts-server/pull/2777. Need to file a Jira ticket if we decide we want to do this.

**Comment by toufali** [comment]
@rhelmer thanks for your replies – all sounds good to me 👍

**Review by toufali — APPROVED** [review]

**Comment by rhelmer** [comment]
> > * There's a readme section (Trigger Breach Alert Email) that involves this utility – does it need updating as part of this PR?
> 
> Thanks for pointing this out - I think it will need updating, but two things make me think we're not ready yet:
> 
>     1. I'm not sure we've implemented enough in V2 to be able to complete the test in that section (have we ported the breach alerts template yet?)

Filed https://mozilla-hub.atlassian.net/browse/MNTOR-1111

>     2. I feel like we should start explicitly defaulting to V2, otherwise it's hard to tell which version the readme refers to

Filed https://mozilla-hub.atlassian.net/browse/MNTOR-1112

**Pull request merged**
